python/pyspark/sql/tests/connect/test_parity_types.py

```diff
@@ -86,7 +86,7 @@ def test_rdd_with_udt(self):
     def test_udt(self):
         super().test_udt()
 
-    @unittest.skip("SPARK-45018: should support CalendarIntervalType")
+    @unittest.skip("SPARK-45026: spark.sql should support datatypes not compatible with arrow")
```

Comment:
```
spark.sql("SELECT make_interval(100, 11, 1, 1, 12, 30, 01.001001)")

...

pyspark.errors.exceptions.connect.UnsupportedOperationException: [UNSUPPORTED_DATATYPE] Unsupported data type "INTERVAL".

```

`spark.sql("SELECT make_interval(100, 11, 1, 1, 12, 30, 01.001001)")` won't work in python client, not because of the newly added `CalendarIntervalType` itself, but because the `spark.sql` internally creates arrow batch while `CalendarIntervalType` is [not arrow-compatible now](https://github.com/apache/spark/blob/3dcee0ccf7e02acd162da652450b85d34dc4e297/sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala#L62-L63).

here needs further investigation, so I create a new ticket for it.